react/ui-components/src/hoc/BoundaryFilter.js

```
@@ -0,0 +1,675 @@
+            ...updatedOptions[childBoundaryType][value?.code],
+            ...uniqueChildren
+          ];
+        }
+      });
+
+      return updatedOptions;
+    });
+
+    // **Update selected values once at the end**
+    setSelectedValuesCodes(updatedSelectedValues.map((item) => item.code))
+    setSelectedValues((prev) => {
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)